### PR TITLE
Add Launch Pad VIA support

### PR DESCRIPTION
**Pull request by luis-godinez**
## Description

- Add VIA support by default.
- Removed most matrix rgb animations to reduce firmware size.
- Update readme to include instructions for configuring Midi on win/mac client.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by drashna** [comment]
You have some merge conflicts here that need to be resolved.  (eg, your branch needs to be updated)

File: keyboards/ungodly/launch_pad/launch_pad.c [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+}
+
+// 0.91" OLED, 128x32 resolution
+#ifdef OLED_DRIVER_ENABLE
+#   define ANIM_SIZE 636 // number of bytes in array, minimize for adequate firmware size, max is 1024
+    oled_rotation_t oled_init_user(oled_rotation_t rotation) {
+        return OLED_ROTATION_180;  // flips the display 180 degrees
+    }
+    layer_state_t layer_state_set_user(layer_state_t state) {
+        static const char PROGMEM gui_layers[][ANIM_SIZE] = {
```
Comment (zvecr) [inline keyboards/ungodly/launch_pad/launch_pad.c:67]: You shouldnt really have `_user` functions at the keyboard level. This makes it impossible for a user to override the behaviour. They either need to be make weak, or moved to the keymap level.

File: keyboards/ungodly/launch_pad/rules.mk [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+AUDIO_ENABLE = no           # Audio output
+MIDI_ENABLE = yes
+OLED_DRIVER_ENABLE = yes
+ENCODER_ENABLE = yes
+RGB_MATRIX_ENABLE = WS2812
+SPACE_CADET_ENABLE = no
+MAGIC_ENABLE = no
+GRAVE_ESC_ENABLE = no
+LTO_ENABLE = yes
+VIA_ENABLE = yes
```
Comment (zvecr) [inline keyboards/ungodly/launch_pad/rules.mk:31]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/ungodly/launch_pad/rules.mk [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+MIDI_ENABLE = yes
+OLED_DRIVER_ENABLE = yes
+ENCODER_ENABLE = yes
+RGB_MATRIX_ENABLE = WS2812
```
Comment (fauxpark) [inline keyboards/ungodly/launch_pad/rules.mk:26]:
```suggestion
RGB_MATRIX_ENABLE = yes
RGB_MATRIX_DRIVER = WS2812
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/ungodly/launch_pad/rules.mk (hunk 3 above)
Comment (luis-godinez) [reply keyboards/ungodly/launch_pad/rules.mk]: This causes compiling issues.

```
quantum/rgb_matrix_drivers.c:33:21: error: 'DRIVER_ADDR_1' undeclared (first use in this function); did you mean 'TIMER_AVR_H'?
     IS31FL3731_init(DRIVER_ADDR_1);
                     ^~~~~~~~~~~~~
                     TIMER_AVR_H
quantum/rgb_matrix_drivers.c:33:21: note: each undeclared identifier is reported only once for each function it appears in
quantum/rgb_matrix_drivers.c: In function 'flush':
quantum/rgb_matrix_drivers.c:87:35: error: 'DRIVER_ADDR_1' undeclared (first use in this function); did you mean 'TIMER_AVR_H'?
     IS31FL3731_update_pwm_buffers(DRIVER_ADDR_1, 0);
                                   ^~~~~~~~~~~~~
                                   TIMER_AVR_H
 [ERRORS]
 |
 |
 |
make[1]: *** [tmk_core/rules.mk:386: .build/obj_ungodly_launch_pad_default/quantum/rgb_matrix_drivers.o] Error 1
Make finished with errors
make: *** [Makefile:584: ungodly/launch_pad:default] Error 1
```

Comment (fauxpark) [reply keyboards/ungodly/launch_pad/rules.mk:26]: Update your repo.
https://github.com/qmk/qmk_firmware/pull/10231

File: keyboards/keebwerk/nano_slider/keymaps/default/keymap.c [hunk 4]
```diff
@@ -74,14 +74,9 @@ bool process_record_user(uint16_t keycode, keyrecord_t *record) {
...
-    if (divisor++) { // only run the slider function 1/256 times it's called
-        return;
-    }
-
-    midi_send_cc(&midi_device, 2, 0x3E, 0x7F - (analogReadPin(SLIDER_PIN) >> 3));
-}
-
 void matrix_scan_user(void) {
-    slider();
+  if (divisor++) { // only run the slider function 1/256 times it's called
```
Comment (fauxpark) [inline keyboards/keebwerk/nano_slider/keymaps/default/keymap.c:78]: Please revert these changes.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by noroadsleft — COMMENTED** [review]
According to [VIA's own docs](https://caniusevia.com/docs/configuring_qmk#create-a-rulesmk-in-keyboardskeyboard_namekeymapsvia), VIA is incompatible with `MIDI_ENABLE = yes` and current versions of QMK (anything version 0.11.0 or newer) have a bug that is yet to be solved which makes keycode assignments not work correctly in VIA anyway.

[[Bug] Keycodes get shifted by one after #9940, that breaks correct indication in VIA · Issue #11157 · qmk/qmk_firmware](https://github.com/qmk/qmk_firmware/issues/11157)

At this point in time I cannot in good faith recommend including VIA support.

**Review by drashna — APPROVED** [review]

**Review by fauxpark — CHANGES_REQUESTED** [review]
.

File: keyboards/keebwerk/nano_slider/keymaps/default/keymap.c (hunk 4 above)
Comment (luis-godinez) [reply keyboards/keebwerk/nano_slider/keymaps/default/keymap.c:78]: reverted in subsequent commit.

**Comment by luis-godinez** [comment]
> According to [VIA's own docs](https://caniusevia.com/docs/configuring_qmk#create-a-rulesmk-in-keyboardskeyboard_namekeymapsvia), VIA is incompatible with `MIDI_ENABLE = yes` and current versions of QMK (anything version 0.11.0 or newer) have a bug that is yet to be solved which makes keycode assignments not work correctly in VIA anyway.
> 
> [[Bug] Keycodes get shifted by one after #9940, that breaks correct indication in VIA · Issue #11157 · qmk/qmk_firmware](https://github.com/qmk/qmk_firmware/issues/11157)
> 
> At this point in time I cannot in good faith recommend including VIA support.

Removed via support until issues caused by #11157 is resolved.

**Review by fauxpark — CHANGES_REQUESTED** [review]
.

File: keyboards/keebwerk/nano_slider/keymaps/default/keymap.c (hunk 4 above)
Comment (Erovia) [reply keyboards/keebwerk/nano_slider/keymaps/default/keymap.c:78]: I see no reason why this PR should touch this file.

File: keyboards/ungodly/launch_pad/readme.md [hunk 5]
```diff
@@ -35,3 +33,63 @@ Make example for this keyboard (after setting up your build environment):
...
+    	</ul>
+    	</td>
+    	<td valign="top">
+        	<ul>
+            	<li><a href="https://rsjaffe.github.io/MIDI2LR/">Midi2Lightroom</a></li>
+            	<li><a href="https://www.orderedbytes.com/controllermate/">ControllerMate</a></li>
+        	</ul>
+    	</td>
+    </tr>
+</table>
```
Comment (Erovia) [inline keyboards/ungodly/launch_pad/readme.md:74]: I think these can be done with Markdown as well without any HTML.

**Review by Erovia — CHANGES_REQUESTED** [review]
